explain-joins.md

```diff
@@ -0,0 +1,299 @@
+
+An index join using the hint [`INL_JOIN`](/optimizer-hints.md#inl_joint1_name--tl_name-) will create a hash table of the intermediate results before joining on the outer table. TiDB also supports creating a hash table on the outer table with the hint [`INL_HASH_JOIN`](/optimizer-hints.md#inl_hash_join). If the column sets on the inner table match the columns of the outer table, the [`INL_MERGE_JOIN`](/optimizer-hints.md#inl_merge_join) index join can apply. Each of these variations of Index Join will be automatically selected by the SQL Optimizer.
+
+### Configuration
+
+Index Join performance is influenced by the following system variables:
+
+* [`tidb_index_join_batch_size`](/system-variables.md#tidb_index_join_batch_size) (default: 25000) - the batch size of index lookup operations.
+* [`tidb_index_lookup_join_concurrency`](/system-variables.md#tidb_index_lookup_join_concurrency) (default: 4) - the number of concurrent index lookup tasks.
+
+## Hash Join
+
+A hash join reads and caches the data on the `Build` side of the join in a hash table, and then reads the data on the `Probe` side of the join, probing the hash table to access required rows. Hash joins require more memory to execute than Index Joins, but execute much faster when there are a lot of rows that need to be joined. The Hash Join operator is multi-threaded in TiDB, and executes in parallel.
+
+An example Hash Join is as follows:
+
+{{< copyable "sql" >}}
+
+```sql
+EXPLAIN SELECT /*+ HASH_JOIN(t1, t2) */ * FROM t1, t2 WHERE t1.id = t2.id;
+```
+
+```sql
++-----------------------------+-----------+-----------+---------------+------------------------------------------------+
+| id                          | estRows   | task      | access object | operator info                                  |
++-----------------------------+-----------+-----------+---------------+------------------------------------------------+
+| HashJoin_27                 | 142020.00 | root      |               | inner join, equal:[eq(test.t1.id, test.t2.id)] |
+| ├─TableReader_29(Build)     | 142020.00 | root      |               | data:TableFullScan_28                          |
+| │ └─TableFullScan_28        | 142020.00 | cop[tikv] | table:t1      | keep order:false                               |
+| └─TableReader_31(Probe)     | 180000.00 | root      |               | data:TableFullScan_30                          |
+|   └─TableFullScan_30        | 180000.00 | cop[tikv] | table:t2      | keep order:false                               |
++-----------------------------+-----------+-----------+---------------+------------------------------------------------+
+5 rows in set (0.00 sec)
+```
+
+The execution process of `Hash Join` is as follows:
+
+1. Cache the data of the `Build` side in memory.
+2. Construct a Hash Table on the `Build` side based on the cached data.
+3. Read the data at the `Probe` side.
+4. Use the data of the `Probe` side to probe the Hash Table.
+5. Return qualified data to the user.
```

Comment:
Are the logical subjects of the 5 sentences (execution process) all `TiDB`?

Reply:
Yes, this is all TiDB.